pyrtools/tools/display.py

```diff
@@ -351,95 +359,123 @@ def check_shape_1d(shapes):
+        subtitle giving the plotted range and dimensionality (with zoom)
+    
         * if `str`, will put the same title on every plot.
-        * if `list`, all values must be `str`, must be the same length as img, assigning each
-          title to corresponding image.
-        * if None, no title will be printed.
-    col_wrap : `int` or None
-        number of axes to have in each row. If None, will fit all axes in a single row.
-    ax : `matplotlib.pyplot.axis` or None
-        if None, make the appropriate figure. otherwise, we resize it so that it's the appropriate
-        number of pixels (done by shrinking the bbox - if the bbox is already too small, this will
-        throw an Exception!, so first define a large enough figure using either make_figure or
+        * if `list`, all values must be `str`, must be the same length as img,
+          assigning each title to corresponding image.
+        * if `None`, no title will be printed (and subtitle will be removed;
+          unsupported for complex tensors).
+    col_wrap : `int` or None, optional
+        number of axes to have in each row. If None, will fit all axes in a
+        single row.
+    ax : `matplotlib.pyplot.axis` or None, optional
+        if None, we make the appropriate figure. otherwise, we resize the axes
+        so that it's the appropriate number of pixels (done by shrinking the
+        bbox - if the bbox is already too small, this will throw an Exception!,
+        so first define a large enough figure using either make_figure or
         plt.figure)
-    cmap : matplotlib colormap
+    cmap : matplotlib colormap, optional
         colormap to use when showing these images
-    plot_complex : {'rectangular', 'polar', 'logpolar'}
+    plot_complex : {'rectangular', 'polar', 'logpolar'}, optional
         specifies handling of complex values.
 
         * `'rectangular'`: plot real and imaginary components as separate images
         * `'polar'`: plot amplitude and phase as separate images
         * `'logpolar'`: plot log_2 amplitude and phase as separate images
-        for any other value, we raise a warning and default to rectangular.
+    as_rgb : bool, optional
+        Whether to plot the image(s) as a grayscale or RGB(A) images. See the
+        `images` for description of what's necessary. Will raise Exception if
+        can't figure out how to plot as specified.
+    kwargs :
+        Passed to `ax.imshow`
 
     Returns
     -------
     fig : `PyrFigure`
         figure containing the plotted images
-    '''
 
+    """
     if plot_complex not in ['rectangular', 'polar', 'logpolar']:
-        warnings.warn("Don't know how to handle plot_complex value %s, defaulting to "
-                      "'rectangular'" % plot_complex)
-        plot_complex = 'rectangular'
+        raise Exception("Don't know how to handle plot_complex value "
+                        f"{plot_complex}!")
 
     try:
```

Comment:
The current code might report a random error if the argument `image` is neither a `list` nor a `np.ndarray`. I'd recommend this:
```python
if type(image) is np.ndarray:
# not needed since it's checked later.
#    if image.ndim not in [2, 3]:
#        raise ValueError("unexpected dimension")
#    else:
    image = [image]

if type(image) is not list:
    raise TypeError('must be a list')    
```

Reply:
Ah, that's a good idea. I wrote it the way I did to try and make it more "duck typing", in case the user passes an object that can be converted easily to an array. Like `torch.tensor`, for example: there are cases where passing a tensor directly to imshow works as you'd like, which the above rules out. But given that we're also implementing `plenoptic.imshow`, which will handle tensors, that's probably no longer necessary...

Reply:
Okay, so I just added this, but I'd like to get input from more people: this means we no longer support tensors at all. That's not bad, because, as I said, we're writing `plenoptic.imshow` to handle this, and tensors didn't always behave like you wanted them to. But it is a major change.